src/boostvqe/boost.py

```diff
@@ -38,41 +37,130 @@
     train_vqe,
 )
 
+NSHOTS = 100
 
-def main(args):
-    """VQE training."""
-    # set backend and number of classical threads
 
-    if args.platform is not None:
-        qibo.set_backend(backend=args.backend, platform=args.platform)
-    else:
-        qibo.set_backend(backend=args.backend)
-        args.platform = GlobalBackend().platform
+def dbqa_vqe(
+    circuit: Circuit,  # in place of ansatz
+    output_folder: str,
+    backend: Optional[str] = "qibojit",
+    platform: Optional[str] = None,
+    optimizer: str = "Powell",
+    optimizer_options: dict = {},
+    tol: float = TOL,
+    decay_rate_lr: float = 1.0,
+    nboost: int = 1,
+    boost_frequency: int = 10,
+    dbi_steps: int = 1,
+    store_h: bool = False,
+    hamiltonian: str = "XXZ",
+    seed: int = SEED,
+    nshots: int = NSHOTS,
+    nlayers: int = 0,
+    mode: DoubleBracketGeneratorType = DoubleBracketGeneratorType.single_commutator,
+):
+    """
+    Args:
+
+       circuit (Circuit): VQE circuit.
+
+       output_folder (str): where to dump the results.
+
+       backend (str, default: "qibojit"):
+            Specifies the quantum backend to use. Options include different backend implementations
+            such as "qibojit" or "qibotf".
+
+        platform (str, default: None):
+            Specifies the platform to run on, typically used for running on GPUs (e.g., "CUDA").
+
+        optimizer (str, default: "Powell"):
+            Optimizer used for VQE training. Options include "Powell", "BFGS", and others.
+
+        optimizer_options (str, optional):
+            Custom options to fine-tune the behavior of the optimizer.
+
+        tol (float, default: TOL):
+            Absolute precision tolerance to stop VQE training once the desired convergence is achieved.
+
+        decay_rate_lr (float, default: 1.0):
+            Learning rate decay factor used when the optimizer is SGD (stochastic gradient descent).
+
+        nboost (int, default: 1):
+            Number of times DBI (Deterministic Boost Iteration) is applied in the optimization process.
+            If set to 1, no boosting is performed.
+
+        boost_frequency (int, default: 10):
+            Number of optimization steps between DBI boosting calls.
 
-    if args.optimizer_options is None:
-        opt_options = {}
+        dbi_steps (int, default: 1):
```

Comment:
```suggestion
        dbqa_steps (int, default: 1):
```
this code should be capable of running DBI (dense matrices) and GCI (qibo.Circuit)